features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/gson/GsonExt.kt

```diff
@@ -6,37 +6,56 @@
 
 package com.datadog.android.sessionreplay.internal.gson
 
+import com.datadog.android.api.InternalLogger
 import com.google.gson.JsonArray
 import com.google.gson.JsonElement
 import com.google.gson.JsonObject
 import com.google.gson.JsonPrimitive
 
+internal const val BROKEN_JSON_ERROR_MESSAGE =
+    "SR GsonExt: Unable parse the batch data into a JsonObject"
+
 @Suppress("SwallowedException")
-internal fun JsonElement.safeGetAsJsonObject(): JsonObject? {
+internal fun JsonElement.safeGetAsJsonObject(internalLogger: InternalLogger): JsonObject? {
     return try {
         asJsonObject
     } catch (e: IllegalStateException) {
-        // TODO: RUMM-2397 Add the proper logs here once the sdkLogger will be added
+        internalLogger.log(
+            InternalLogger.Level.ERROR,
+            InternalLogger.Target.TELEMETRY,
+            { BROKEN_JSON_ERROR_MESSAGE },
+            e
+        )
         null
     }
 }
 
 @Suppress("SwallowedException")
-internal fun JsonPrimitive.safeGetAsLong(): Long? {
+internal fun JsonPrimitive.safeGetAsLong(internalLogger: InternalLogger): Long? {
```

Comment:
Same comments as above

Reply:
so it is my bad here as I was not aware that we are not logging the throwable stacktrace in telemetry :(. In that case I think the throwable message would make sense along with the Log message. What do you think ?